lib/workos/passwordless.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+# typed: true
+
+require 'net/http'
+require 'uri'
+
+module WorkOS
+  # The Passwordless module provides convenience methods for working with
+  # passwordless sessions including the WorkOS Magic Link. You'll need a valid
+  # API key.
+  #
+  # @see https://workos.com/docs/sso/configuring-magic-link
+  module Passwordless
+    class << self
+      extend T::Sig
+      include Base
+      include Client
+
+      # Create an Passwordless Session.
+      #
+      # @param [Hash] options A hash with options for the session
+      # @option options [String] email The email of the user to authenticate.
+      # @option options [String] state Optional parameter that the redirect URI
+      #  received from WorkOS will contain. The state parameter can be used to
+      #  encode arbitrary information to help restore application state between
+      #  redirects.
+      # @option options [String] type The type of Passwordless Session to
+      #  create. Currently, the only supported value is 'MagicLink'.
+      #
+      # @return Hash
+      sig do
+        params(
+          options: Hash,
+        ).returns(::T.untyped)
```

Comment:
I wonder if we want to make a `PasswordlessSessionStruct` and return that. See an example in `OrganizationStruct`

Reply:
- [ ] Add `PasswordlessSessionStruct`